File: src/com/leetcode/Easy/MergeTwoSortedLists.java
package com.leetcode.Easy;

// 21. Merge Two Sorted Lists
public class MergeTwoSortedLists {
    static class ListNode {
        int val;
        ListNode next;

        ListNode(int value) {
            val = value;
            next = null;
        }
    }

    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        ListNode newHead = new ListNode(0);
        ListNode node = newHead;
        if (l1 == null) return l2;
        else if (l2 == null) return l1;
        else {
            while (l1 != null && l2 != null) {
                if (l1.val <= l2.val) {
                    node.next = l1;
                    l1 = l1.next;
                } else {
                    node.next = l2;
                    l2 = l2.next;
                }
                node = node.next;
            }
            while (l1 != null) {
                node.next = l1;
                l1 = l1.next;
                node = node.next;
            }
            while (l2 != null) {
                node.next = l2;
                l2 = l2.next;
                node = node.next;
            }


            return newHead.next;
        }
    }
}

File: JavaScript/Easy/MergeTwoSortedLists.js
/**
 * 21. Merge Two Sorted Lists
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} list1
 * @param {ListNode} list2
 * @return {ListNode}
 */
var mergeTwoLists = function (list1, list2) {
  let temp = new ListNode(0);
  let curr = temp;
  while (list1 != null && list2 != null) {
    if (list1.val < list2.val) {
      curr.next = new ListNode(list1.val);
      list1 = list1.next;
    } else {
      curr.next = new ListNode(list2.val);
      list2 = list2.next;
    }

    curr = curr.next;
  }

  while (list1 != null) {
    curr.next = new ListNode(list1.val);
    list1 = list1.next;
    curr = curr.next;
  }

  while (list2 != null) {
    curr.next = new ListNode(list2.val);
    list2 = list2.next;
    curr = curr.next;
  }

  return temp.next;
};
